neuralmonkey/util/word2vec.py

```diff
@@ -0,0 +1,96 @@
+"""Word2vec plug-in module.
+
+This module provides functionality needed to work with word2vec files.
+"""
+
+from typing import Callable, List
+
+import numpy as np
+from typeguard import check_argument_types
+
+from neuralmonkey.vocabulary import (
+    Vocabulary, is_special_token, SPECIAL_TOKENS)
+
+
+class Word2Vec:
+
+    def __init__(self, path: str, encoding: str = "utf-8") -> None:
+        """Load the word2vec file."""
+        check_argument_types()
+
+        # Create the vocabulary object, load the words and vectors from the
+        # file
+
+        self.vocab = Vocabulary()
+        embedding_vectors = []  # type: List[np.ndarray]
+        emb_size = None
+
+        with open(path, encoding=encoding) as f_data:
+            for line in f_data:
+                fields = line.split()
+                word = fields[0]
+                vector = np.fromiter((float(x) for x in fields[1:]),
+                                     dtype=np.float)
+
+                if emb_size is None:
+                    emb_size = int(fields[1])
+
+                    # Add zero embeddings for padding, start, and end token
+                    embedding_vectors.append(np.zeros(emb_size))
+                    embedding_vectors.append(np.zeros(emb_size))
+                    embedding_vectors.append(np.zeros(emb_size))
+                    # Add placeholder for embedding of the unknown symbol
+                    embedding_vectors.append(None)
+                    continue
```

Comment:
Bylo by **mnohem** čitelnější nejdřív zpracovat první řádek zvlášť a pak pustit ten cyklus na zbytek, než mít uprostřed cyklu část, která je stejně dlouhá jako jeho zbytek a provede se jenom jednou...

Reply:
Nebo ten `if` aspoň hodit hned za řádek 30, aby to nevypadalo, že se tam z první řádky vyrábí vektor.

Reply:
jasný, udělal jsem to--